src/fides/api/models/attachment.py

```diff
@@ -183,16 +205,34 @@ def retrieve_attachment(
             )
             return size, url
 
+        if self.config.type == StorageType.gcs:
+            bucket_name = f"{self.config.details[StorageDetails.BUCKET.value]}"
+            auth_method = self.config.details[StorageDetails.AUTH_METHOD.value]
+            storage_client = get_gcs_client(auth_method, self.config.secrets)
+            bucket = storage_client.bucket(bucket_name)
+            blob = bucket.blob(f"{self.id}/{self.file_name}")
+
+            # Ensure we have the blob metadata
+            blob.reload()
+
+            url = blob.generate_signed_url(
+                version="v4",
+                expiration=CONFIG.security.subject_request_download_link_ttl_seconds,
+                method="GET",
+            )
+            return blob.size, url
+
         if self.config.type == StorageType.local:
             filename = get_local_filename(f"{self.id}/{self.file_name}")
             with open(filename, "rb") as file:
-                size = len(file.read())
+                content = file.read()
+                size = len(content)
                 return size, filename
```

Comment:
We could do something like this to avoid having to load the entire file into memory
```suggestion
                size = os.path.getsize(filename)
                return size, filename
```